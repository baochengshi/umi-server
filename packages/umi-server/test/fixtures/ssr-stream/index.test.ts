import { join } from 'path';
import server from '../../../src';

test('ssr stream', async () => {
  const render = server({
    root: join(__dirname, 'dist'),
    stream: true,
  });
  const { ssrStream } = await render({
    req: {
      url: '/',
    },
  });
  console.log('ssrStream', ssrStream);
  expect(ssrStream._str).toMatchSnapshot();
});
