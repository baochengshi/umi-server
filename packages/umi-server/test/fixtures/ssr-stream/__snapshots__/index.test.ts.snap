// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr stream 1`] = `"<!DOCTYPE html><html data-reactroot><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no\\"><script>window.g_useSSR=true;</script><script>window.routerBase = \\"/\\";</script></head><body><div id=\\"root\\"><div class=\\"wrapper\\"><h1>Hello UmiJS SSR</h1><ul><li>Alice</li><li>Jack</li><li>Tony</li></ul><button>0</button></div></div><script>window.g_initialData = {\\"list\\":[{\\"name\\":\\"Alice\\"},{\\"name\\":\\"Jack\\"},{\\"name\\":\\"Tony\\"}]};</script><script src=\\"/umi.js\\"></script></body></html>"`;
